### ISSUE TITLE
[ORM] Hierarchy extension error

### ISSUE THREAD

**Issue by a2nt** [silverstripe/silverstripe-framework#6127]
Code bellow will return SQL error when Children() will be required cuz method will try to get ParentID field of ModalObject table.

``` php
class ModalObject extends DataObject {
   private static $db = [....];
}

class ServiceType extends ModalObject {
  private static $db = [....];
  private static $extensions = [
     'Hierarchy',
  ];
}
```

**Comment by dhensby** [comment]
@a2nt which version?

**Comment by dhensby** [comment]
OK - seen it in 3.4.x-dev

The problem is `Hierarchy` is specifically querying against the base table (not sure why). - https://github.com/silverstripe/silverstripe-framework/blob/3.4/model/Hierarchy.php#L656

The error goes away when you just use the class of the current owner. more thorough testing is needed, though.

**Comment by maxime-rainville** [comment]
Looks like where still querying the base class instead of querying the owner class. So this is probably still a problem in SS4.

https://github.com/silverstripe/silverstripe-framework/blob/4b5d2c3cc292a690b5de9c5dec6ac9afe52abe69/src/ORM/Hierarchy/Hierarchy.php#L422-L432

**Comment by michalkleiner** [comment]
Attempted a fix — https://github.com/silverstripe/silverstripe-framework/pull/10077. Need to wait until the tests are running/passing again.

**Comment by michalkleiner** [comment]
Resolved by #10077 and #10107.
